analyzers/src/SonarAnalyzer.CSharp/Rules/MethodParameterUnusedCodeFix.cs

```diff
@@ -19,39 +19,68 @@
+public sealed class MethodParameterUnusedCodeFix : SonarCodeFix
 {
-    [ExportCodeFixProvider(LanguageNames.CSharp)]
-    public sealed class MethodParameterUnusedCodeFix : SonarCodeFix
+    private const string Title = "Remove unused parameter";
+
+    public override ImmutableArray<string> FixableDiagnosticIds => ImmutableArray.Create(MethodParameterUnused.DiagnosticId);
+
+    protected override async Task RegisterCodeFixesAsync(SyntaxNode root, SonarCodeFixContext context)
     {
-        private const string Title = "Remove unused parameter";
+        var diagnostic = context.Diagnostics.First();
+        var diagnosticSpan = diagnostic.Location.SourceSpan;
+        var parameter = root.FindNode(diagnosticSpan, getInnermostNodeForTie: true) as ParameterSyntax;
+        IEnumerable<SyntaxNode> arguments = [];
 
-        public override ImmutableArray<string> FixableDiagnosticIds => ImmutableArray.Create(MethodParameterUnused.DiagnosticId);
+        if (!bool.Parse(diagnostic.Properties[MethodParameterUnused.IsRemovableKey]))
+        {
+            return;
+        }
 
-        protected override Task RegisterCodeFixesAsync(SyntaxNode root, SonarCodeFixContext context)
+        if (parameter?.Parent.Parent is BaseMethodDeclarationSyntax methodDeclaration)
         {
-            var diagnostic = context.Diagnostics.First();
-            var diagnosticSpan = diagnostic.Location.SourceSpan;
-            var parameter = root.FindNode(diagnosticSpan, getInnermostNodeForTie: true) as ParameterSyntax;
+            var model = await context.Document.GetSemanticModelAsync(context.CancellationToken);
+            var parameterIndex = methodDeclaration.ParameterList.Parameters.IndexOf(parameter);
 
-            if (!bool.Parse(diagnostic.Properties[MethodParameterUnused.IsRemovableKey]))
+            if (parameterIndex >= 0 && model.GetDeclaredSymbol(methodDeclaration, context.CancellationToken) is { } methodSymbol)
             {
-                return Task.CompletedTask;
-            }
+                var callers = await SymbolFinder.FindCallersAsync(
+                    methodSymbol,
+                    context.Document.Project.Solution,
+                    ImmutableHashSet.Create(context.Document),
+                    context.CancellationToken);
 
-            context.RegisterCodeFix(
-                Title,
-                c =>
-                {
-                    var newRoot = root.RemoveNode(
-                        parameter,
-                        SyntaxRemoveOptions.KeepLeadingTrivia | SyntaxRemoveOptions.AddElasticMarker);
-                    return Task.FromResult(context.Document.WithSyntaxRoot(newRoot));
-                },
-                context.Diagnostics);
-
-            return Task.CompletedTask;
+                arguments = callers
+                    .SelectMany(x => x.Locations)
+                    .Select(x => GetArgumentFromLocation(x, parameterIndex));
+            }
```

Comment:
If you move it to a method you can call it directly from the `RemoveNodes` invocation.